processor/src/test/java/org/mapstruct/ap/test/subclassmapping/DeepCloneMethodMapper.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.subclassmapping;
+
+import org.mapstruct.BeanMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.NullValueMappingStrategy;
+import org.mapstruct.SubclassMapping;
+import org.mapstruct.ap.test.subclassmapping.mappables.Bike;
+import org.mapstruct.ap.test.subclassmapping.mappables.Car;
+import org.mapstruct.ap.test.subclassmapping.mappables.Vehicle;
+import org.mapstruct.control.DeepClone;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface DeepCloneMethodMapper {
+    DeepCloneMethodMapper INSTANCE = Mappers.getMapper( DeepCloneMethodMapper.class );
+
+    @SubclassMapping( source = Car.class, target = Car.class )
+    @SubclassMapping( source = Bike.class, target = Bike.class )
+    @BeanMapping( mappingControl = DeepClone.class, nullValueMappingStrategy = NullValueMappingStrategy.RETURN_NULL )
```

Comment:
Why is the `nullValueMappingStrategy` needed here?

Reply:
Because it is not allowed to only define `mappingControl` at the moment:
`Compilation failed. Diagnostics: [DiagnosticDescriptor: ERROR DeepCloneMethodMapper.java:24 'nullValueMappingStrategy', 'nullValuePropertyMappingStrategy', 'resultType' and 'qualifiedBy' are undefined in @BeanMapping, define at least one of them.]]`

Do we want to fix that as well with this PR?

Reply:
Got it, no need to fix it in this. I've created #3040 to fix that.